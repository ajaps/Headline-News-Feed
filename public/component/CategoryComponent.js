import React from 'react';

import SourcesStore from '../stores/Sources';
import * as myActions from '../actions/HeadlineActions';
/**
 * Represents a Source.
 */
export default class CategoryComponent extends React.Component {

  /**
 * sends an action to change category based on user selection
 * @param {String} category The selected category
 * @return {void}
 */
  setCategory(category) {
    const url = SourcesStore.getSourceUrl;
    myActions.setCategory(url, category);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    return (
      <li>
        <a onClick={this.setCategory.bind(this, this.props.name)}>
          {this.props.name}
        </a>
      </li>
    );
  }
}
