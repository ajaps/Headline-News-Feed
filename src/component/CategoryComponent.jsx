import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import SourcesStore from '../stores/Sources';
import * as myActions from '../actions/HeadlineActions';
/**
 * Represents a Source.
 */
export default class CategoryComponent extends React.Component {

  /**
 * sends an action to change category based on user selection
 * @param {String} category The selected category
 * @return {void}
 */
  setCategory(category) {
    const url = SourcesStore.getSourceUrl;
    myActions.setCategory(url, category);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    // sets class name based on user source selection
    const btnClass = classNames({
      active: this.props.category[0] === this.props.id,
      catHighlight: this.props.category[0] === this.props.id,
    });

    return (
      <li className={btnClass}>
        <a onClick={this.setCategory.bind(this, this.props.id)}>
          {this.props.name}
        </a>
      </li>
    );
  }
}

CategoryComponent.propTypes = {
  name: PropTypes.string.isRequired,
};

CategoryComponent.defaultProps = {
  name: '',
};
