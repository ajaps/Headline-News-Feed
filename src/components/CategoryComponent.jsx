import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import SourcesStore from '../stores/Sources';
import * as myActions from '../actions/HeadlineActions';
/**
 * Represents a Source.
 */
export default class CategoryComponent extends React.Component {
  constructor(props) {
    super(props);
    this.setCategory = this.setCategory.bind(this, props.id);
  }


  /**
 * sends an action to change category based on user selection
 * @param {String} category The selected category
 * @return {void}
 */
  setCategory(category) {
    const url = SourcesStore.getSourceUrl;
    myActions.setCategory(url, category);
  }

/**
 * @returns {component} A component with relevant source.
 */
  render() {
    // sets class name based on user source selection
    const btnClass = classNames({
      active: this.props.category[0] === this.props.id,
    });

    return (
      <li className={btnClass}>
        <a onClick={this.setCategory}>
          {this.props.name}
        </a>
      </li>
    );
  }
}

CategoryComponent.propTypes = {
  name: PropTypes.string.isRequired,
  category: PropTypes.array,
  id: PropTypes.string.isRequired,
};

CategoryComponent.defaultProps = {
  name: '',
  category: [],
};
