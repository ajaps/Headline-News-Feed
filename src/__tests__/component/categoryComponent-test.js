import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import CategoryComponent from '../../components/CategoryComponent.jsx';

jest.mock('../../api/NewsApi', () => ({
  getData: () => Promise.resolve('getPromise')
}));

describe('Category', () => {
  let app;
  const onButtonClick = sinon.spy();
  // // const mockFn = jest.fn();
  // // mockFn();
  // // expect(mockFn).toHaveBeenCalled();
  beforeEach(() => {
    // const fetch = jest.fn();
    const categoryItem = { id: 'games', name: 'Games' };
    app = shallow(<CategoryComponent onClick={onButtonClick}
      key={categoryItem.id}{...categoryItem} category={['games']} />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should have extacly one anchor element', () => {
    expect(app.find('a').length).toEqual(1);
  });

  it('should have a displayed text called Games', () => {
    expect(app.text()).toBe('Games');
  });

  it('should call the setCategory function on click the anchor element', () => {
    app.find('a').simulate('click');
    expect(app).toBeTruthy();
  });
});
