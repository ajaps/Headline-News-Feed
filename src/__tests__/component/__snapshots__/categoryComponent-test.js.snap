// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should render as expected 1`] = `
<li
  className="active"
>
  <a
    onClick={[Function]}
  >
    Games
  </a>
</li>
`;
